chingu-voyages/v51-tier3-team-33
# Olga/feature/add friend frontend and email send functionality

# Description

Created form to send an invite to a friend to join the app.


## Main changes explained:
- Created api/email/route.js for email sending
- Created components for AddFriend form. This form can be accessed by clicking on AddFriend button in the sidebar
- Also added some fixes to the UI along the way (no need to test these, they are here and there and only affect UI)


## How to test:
1. check into current branch
2. do `pnpm install` and `pnpm run dev` to run this PR locally
3. in  your .env file add env vars I'll post in developer's channel
4. Log in to the app
5. click on 'Add friend' button on the sidebar
6. verify that this  form opens:

![Screenshot 2024-10-11 at 12 13 33 PM](https://github.com/user-attachments/assets/cc05822c-475d-4a25-bc74-eed0db28d167)

7. Click on 'copy invite link' - a success modal will pop up and you can try pasting the copied link somewhere
8. Click on 'send' without filling in the form and only filling out one of the inputs. Verify that a red error popup message appears
9. Fill out the inputs and click 'cancel'. Verify that the form is cleared.
10. FIll out the form with your real email and send it. Check your inbox and see if you've got an email!

## Note:
Right now the shareable link leads to /login. Not sure whether this is correct, since the user is supposed to become friends with the invitor.
